### PR TITLE
Feature/sleep detection

### PR DESCRIPTION
**Pull request by israeljacob**

### CODE REVIEW

File: blinked_detector.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    :param rgb_image: A NumPy array representing a color image in RGB format.
+    :return: A NumPy array representing the grayscale version of the input image.
+    """
+    gray_image = cvtColor(rgb_image, COLOR_RGB2GRAY)
+    return gray_image
+
+
+def get_face_rects(image: ndarray) -> Optional[rectangle]:
+    """
+    Detect faces in an image using a face detection model.
```
Comment (dsl96) [inline blinked_detector.py]: detect only the largest face

File: blinked_detector.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+             as (left, top, right, bottom) coordinates.
+    """
+    rects = FACE_DETECTOR(image, UP_SAMPLING)
+    if not rects:
+        return None
+    largest_face = max(rects, key=lambda rect: (rect.right() - rect.left()) * (rect.bottom() - rect.top()))
+    return largest_face
+
+
+def get_face_shape(gray_image: ndarray, largest_face: rectangle) -> ndarray:
```
Comment (dsl96) [inline blinked_detector.py]: get_face_shape vs get_face_rects  ???

File: blinked_detector.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    """
+    Get facial landmarks for the largest detected face in a grayscale image.
```
Comment (dsl96) [inline blinked_detector.py]: its not most get the  largest face

File: sleep_deector.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+def handle_counter(counter: int, is_blinked: bool, start_time: float) -> (int, float, bool):
+    """
+    Update a counter based on whether an eye blink is detected and a timing threshold.
+
+    :param counter: An integer representing the current count. :param is_blinked: A boolean indicating whether an eye
+    blink is detected. :param start_time: A float representing the start time of the measurement. :return: A tuple
+    containing the updated counter (int) and the updated start time (float) and a flag that says if half a second has
+    passed (bool).
+    """
+    if is_blinked and time() - start_time > 0.5:
```
Comment (dsl96) [inline sleep_deector.py]: 0.5 hard code

**Review by dsl96 — CHANGES_REQUESTED** [review]

File: blinked_detector.py (hunk 1 above)
Comment (israeljacob) [reply blinked_detector.py]: Solved

File: blinked_detector.py (hunk 2 above)
Comment (israeljacob) [reply blinked_detector.py]: Solved

File: sleep_deector.py (hunk 4 above)
Comment (israeljacob) [reply sleep_deector.py]: Solved

**Pull request merged**